### DISCUSSION TITLE
Support _any_ implementation of the `EventTarget` interface for `addEventListener`

### DISCUSSION THREAD

**Discussion by dancrumb** [juliencrn/usehooks-ts#555] [Ideas]
First: I love this package!

Second, I was a little surprised to learn that `addEventListener` only supports DOM elements, the window or a media query list as `element`.

Looking through the code, I couldn't immediately see why it couldn't also support any instance of the `EventTarget` interface.

I'm happy to create a PR, but I wanted to see if there was a pre-existing reason for the current support limit.

**Comment by juliencrn** [comment]
Thanks! I'm curious to see your proposal.

**Comment by dancrumb** [comment]
I created #585, since I think an implementation is probably the clearest proposal :)

Happy to modify, refine, etc.
